jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -158,19 +158,12 @@
      * has its session completed as the request exits the context.
      */
     public class SessionAsyncListener implements AsyncListener
-    {
-        private Session _session;
-        
-        public SessionAsyncListener(Session session)
-        {
-           _session = session;
-        }
-
+    {        
         @Override
         public void onComplete(AsyncEvent event) throws IOException
         {
             //An async request has completed, so we can complete the session
-            complete(((HttpServletRequest)event.getAsyncContext().getRequest()).getSession(false));
+            complete(Request.getBaseRequest(event.getAsyncContext().getRequest()).getSession(false));
```

Comment:
wouldn't it make sense to just have a `getJettySession(HttpServletRequest request)` generic method that you use everywhere?